### PR TITLE
improve Caddy & Compose files for easier dev setups

### PR DESCRIPTION
**Pull request by Zocker1999NET**
for easier setting up dev environments:

- use env vars for caddy domains (now overwritable using a `.env` file)
- add build instructions to backend & frontend images into docker-compose.yml

I have tested those "on the field" while working on #34. When not given explicit intruction to build, docker compose should divert to downloading the images from GitHub.
And Caddy (or rather Docker Compose) should default the DOMAIN_LONG & DOMAIN_SHORT variables to smd-karlsruhe.de & smd-ka.de respectively.

### CODE REVIEW

**Review by Techthy — APPROVED** [review]
Works on the server. Good to go. Just out of curiosity: how does the name:- syntax work tho?

**Comment by Zocker1999NET** [comment]
> Works on the server. Good to go. Just out of curiosity: how does the name:- syntax work tho?

That’s called interpolation, see [docs](https://docs.docker.com/compose/compose-file/12-interpolation/), Docker Compose hereby uses a similar syntax than most shell (e.g. bash) do.

TL;DR: `:-` allows to define a default which gets used if the named variable is either not set or empty.

**Comment by Zocker1999NET** [comment]
I will merge this PR today afternoon, so I have some free time, just in case
